pkg/ostreeuploader/push.go

```diff
@@ -120,19 +116,20 @@ func (t FioToken) SetAuthHeader(req *http.Request) {
 	req.Header.Set("osf-token", string(t))
 }
 
-func newOSTreeHubAccessorWithToken(repo string, factory string, token string, apiVer string) (*ostreeHubAccessor, error) {
+func newOSTreeHubAccessorWithToken(repo, hubURL, factory, token, apiVer string) (*ostreeHubAccessor, error) {
 	if err := checkRepoDir(repo); err != nil {
 		return nil, err
 	}
 	hub := &OSTreeHub{
-		URL:     FioApiBaseUrl,
+		URL:     hubURL,
```

Comment:
I think it shouldn't impact customer since we still use the creds archive in meta-lmp?

Reply:
that was my thinking. There will be work to use this correctly in ci-scripts.

Reply:
There is also this doc https://docs.foundries.io/latest/user-guide/custom-ci/custom-ci-for-rootfs.html#push-ostree-repo-to-cloud that most likely is impacted by this change.

Maybe we should default to `"https://api.foundries.io"` if `hubURL` is not specified.

Reply:
i think i may have messed that up. i'll test this more today

Reply:
I did just do a fixup change to make our URL concatenation/joining logic safer.